auditbeat/module/file_integrity/kprobes/events_cache.go

```diff
@@ -0,0 +1,156 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build linux
+
+package kprobes
+
+import (
+	"path/filepath"
+)
+
+type dEntriesIndex map[dKey]*dEntry
+type dEntriesMoveIndex map[uint64]*dEntry
+
+type dEntryCache struct {
```

Comment:
do we need locks for remove*, add*, move*?

Reply:
hmmm I thought I had a comment about it but nope is not there and I will add it to keep track of it; dEntryCache is expected to be handle only by a single goroutine at least in this iteration. So a locking mechanism is not required at this point at least